### PR TITLE
Offline detector

### LINKED ISSUE #409: Offline detector

### PR DESCRIPTION
**Pull request by Katrin-kudryash**
## PR includes

-   [x] Feature

## Related issues

Resolve #409

### CODE REVIEW

**Review by nyet-ty — APPROVED** [review]

File: src/components/OfflineBanner/OfflineBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { onlineManager } from '@tanstack/react-query';
+import styled from 'styled-components';
+import { useOfflineDetector } from '@taskany/bricks';
+
+import { tr } from './OfflineBanner.i18n';
+
+const StyledBanner = styled.div`
+    width: 100%;
+    min-height: 40px;
```
Comment (awinogradov) [inline src/components/OfflineBanner/OfflineBanner.tsx:9]: `heigh: 40px`

File: src/components/OfflineBanner/OfflineBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    max-height: 40px;
+    padding-top: 10px;
```
Comment (awinogradov) [inline src/components/OfflineBanner/OfflineBanner.tsx:11]: use gap from `@taskany/colors`

File: src/components/OfflineBanner/OfflineBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    background-color: #d24b4e;
```
Comment (awinogradov) [inline src/components/OfflineBanner/OfflineBanner.tsx:12]: use danger color from `@taskany/colors`

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/components/OfflineBanner/OfflineBanner.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    text-align: center;
+`;
+
+export const OfflineBanner = () => {
+    const online = useOfflineDetector({
+        setStatus: () => onlineManager.setOnline(),
+        remoteServerUrl: '/api/health',
+    });
+    return online ? null : <StyledBanner>{tr('You are currently offline. Check connection.')}</StyledBanner>;
```
Comment (tsumo) [inline src/components/OfflineBanner/OfflineBanner.tsx:21]: nullable

**Review by tsumo — CHANGES_REQUESTED** [review]

**Comment by Katrin-kudryash** [comment]
Untitled

**Review by awinogradov — APPROVED** [review]

**Review by tsumo — APPROVED** [review]

**Comment by Katrin-kudryash** [comment]
Untitled

**Review by 9teen90nine — APPROVED** [review]

**Review by awinogradov — CHANGES_REQUESTED** [review]
telegram-cloud-document-2-5326033289859842415

**Comment by Katrin-kudryash** [comment]
Untitled 3

**Comment by Katrin-kudryash** [comment]
Untitled 2

File: src/components/OfflineBanner/OfflineBanner.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            onlineManager.setOnline(globalOnlineStatus || remoteServerStatus);
+        },
+        remoteServerUrl: '/api/health',
+    });
+    return nullable(!globalOnlineStatus || !remoteServerStatus, () => (
+        <StyledBanner>
+            <IconExclamationCircleOutline size="s" color="#f7c0be" />
+            <Text as="span" color="#f7c0be">
+                {tr('You are currently offline. Check connection.')}
+            </Text>{' '}
```
Comment (9teen90nine) [inline src/components/OfflineBanner/OfflineBanner.tsx:33]: what's this?

**Review by awinogradov — APPROVED** [review]
@Katrin-kudryash @9teen90nine we need ticket to move this component to bricks and use everywhere

**Comment by 9teen90nine** [comment]
> @Katrin-kudryash @9teen90nine we need ticket to move this component to bricks and use everywhere

https://github.com/taskany-inc/bricks/issues/806

**Pull request merged**